File: app/src/main/java/afkt/project/ui/adapter/ButtonAdapter.java
package afkt.project.ui.adapter;

import android.support.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

import afkt.project.R;
import afkt.project.model.item.ButtonValue;

/**
 * detail: Button 适配器
 * @author Ttt
 */
public class ButtonAdapter extends BaseQuickAdapter<ButtonValue, BaseViewHolder> {

    public ButtonAdapter(@Nullable List<ButtonValue> data) {
        super(R.layout.base_view_button, data);
    }

    @Override
    protected void convert(BaseViewHolder helper, ButtonValue item) {
        helper.setText(R.id.vid_bvb_btn, item.text)
                .addOnClickListener(R.id.vid_bvb_btn)
                .addOnLongClickListener(R.id.vid_bvb_btn);
    }
}

File: app/src/main/java/afkt/project/ui/adapter/ButtonAdapter.kt
package afkt.project.ui.adapter

import afkt.project.R
import afkt.project.databinding.BaseViewButtonBinding
import afkt.project.model.item.ButtonValue
import android.view.ViewGroup
import dev.adapter.DevDataAdapterExt
import dev.base.adapter.DevBaseViewBindingVH
import dev.base.adapter.newBindingViewHolder

/**
 * detail: Button 适配器
 * @author Ttt
 */
class ButtonAdapter(data: List<ButtonValue>) : DevDataAdapterExt<ButtonValue, DevBaseViewBindingVH<BaseViewButtonBinding>>() {

    init {
        setDataList(data, false)
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): DevBaseViewBindingVH<BaseViewButtonBinding> {
        return newBindingViewHolder(parent, R.layout.base_view_button)
    }

    override fun onBindViewHolder(
        holder: DevBaseViewBindingVH<BaseViewButtonBinding>,
        position: Int
    ) {
        val item = getDataItem(position)
        holder.binding.vidBvbBtn.apply {
            text = item.text
            setOnClickListener {
                mItemCallback?.onItemClick(item, position)
            }
            setOnLongClickListener {
                mItemCallback?.onItemLongClick(item, position)
                true
            }
        }
    }
}